test/gtest/core/test_mc.cc

```diff
@@ -29,14 +52,36 @@ UCC_TEST_F(test_mc, init_is_required)
 
 UCC_TEST_F(test_mc, can_alloc_and_free_host_mem)
 {
+	// mpool will be used only if size is smaller than UCC_MC_CPU_ELEM_SIZE, which by default set to 1024 and is configurable at runtime.
+	size_t size = 4096;
+    ucc_mc_buffer_header_t *h    = NULL;
     void *ptr = NULL;
-    size_t size = 4096;
 
     ASSERT_EQ(UCC_OK, ucc_constructor());
     ASSERT_EQ(UCC_OK, ucc_mc_init());
-    EXPECT_EQ(UCC_OK, ucc_mc_alloc(&ptr, size, UCC_MEMORY_TYPE_HOST));
+    EXPECT_EQ(UCC_OK, ucc_mc_alloc(&h, size, UCC_MEMORY_TYPE_HOST));
+    ptr = h->addr;
     memset(ptr, 0, size);
-    EXPECT_EQ(UCC_OK, ucc_mc_free(ptr, UCC_MEMORY_TYPE_HOST));
+    EXPECT_EQ(UCC_OK, ucc_mc_free(h, UCC_MEMORY_TYPE_HOST));
+    ucc_mc_finalize();
+}
+
+UCC_TEST_F(test_mc, can_alloc_and_free_host_mem_mt)
+{
+	// mpool will be used only if size is smaller than UCC_MC_CPU_ELEM_SIZE, which by default set to 1024 and is configurable at runtime.
+	size_t size = 1024;
+    int num_of_threads = 20;
+
+    std::vector<pthread_t> threads;
+    threads.resize(num_of_threads);
+    ASSERT_EQ(UCC_OK, ucc_constructor());
+    ASSERT_EQ(UCC_OK, ucc_mc_init());
+    for (int i = 0; i < num_of_threads; i++) {
+    	pthread_create(&threads[i], NULL, &mt_ucc_mc_cpu_calls, (void *)&size);
```

Comment:
Maybe to generalize the test we can pass different sizes to different threads. E.g. size = 64, and you pass size = size*2 to each thread, so first thread runs 64 bytes and thread 20 will run 64*2**20, ie. 64 Mb.